### ISSUE TITLE
$bits function returns 0 when struct is located outside of current package.

### ISSUE THREAD

**Issue by QuantamHD** [chipsalliance/Surelog#2537]
```
package b;
  typedef struct packed {
    logic                one;
    logic [10:0]         vc;
    logic [10:0]         vb;
  } foo_bar_t;

endpackage : b

package aa;

  function automatic b::foo_bar_t foo_bar;

    input b::foo_bar_t a;

    b::foo_bar_t y;
    reg [$bits(a)-1:0] a_vec;

    begin
      foo_bar = y; // return the whole struct
    end

  endfunction : foo_bar
endpackage : aa
```

$bits(a) is returned incorrectly as 0 when it should return 23.

**Comment by QuantamHD** [comment]
@hzeller One problem I found in our internal project

**Comment by alaindargelas** [comment]
Fixed by #2539
